pip install backports-datetime-fromisoformat no longer works

I'm on a Mac (OSX 15.1), but this behavior has been [observed on Fedora Linux](https://github.com/eprbell/dali-rp2/issues/273) as well.

When I try:
`pip install backports-datetime-fromisoformat`

I get:
```
Collecting backports-datetime-fromisoformat
  Using cached backports_datetime_fromisoformat-2.0.2.tar.gz (23 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Building wheels for collected packages: backports-datetime-fromisoformat
  Building wheel for backports-datetime-fromisoformat (pyproject.toml) ... error
  error: subprocess-exited-with-error
  
  × Building wheel for backports-datetime-fromisoformat (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [32 lines of output]
      running bdist_wheel
      running build
      running build_py
      creating build/lib.macosx-15.0-arm64-cpython-313/backports
      copying backports/__init__.py -> build/lib.macosx-15.0-arm64-cpython-313/backports
      creating build/lib.macosx-15.0-arm64-cpython-313/backports/datetime_fromisoformat
      copying backports/datetime_fromisoformat/__init__.py -> build/lib.macosx-15.0-arm64-cpython-313/backports/datetime_fromisoformat
      running egg_info
      writing backports_datetime_fromisoformat.egg-info/PKG-INFO
      writing dependency_links to backports_datetime_fromisoformat.egg-info/dependency_links.txt
      writing top-level names to backports_datetime_fromisoformat.egg-info/top_level.txt
      reading manifest file 'backports_datetime_fromisoformat.egg-info/SOURCES.txt'
      reading manifest template 'MANIFEST.in'
      adding license file 'LICENSE'
      writing manifest file 'backports_datetime_fromisoformat.egg-info/SOURCES.txt'
      copying backports/datetime_fromisoformat/_datetimemodule.c -> build/lib.macosx-15.0-arm64-cpython-313/backports/datetime_fromisoformat
      copying backports/datetime_fromisoformat/_datetimemodule.h -> build/lib.macosx-15.0-arm64-cpython-313/backports/datetime_fromisoformat
      copying backports/datetime_fromisoformat/module.c -> build/lib.macosx-15.0-arm64-cpython-313/backports/datetime_fromisoformat
      copying backports/datetime_fromisoformat/timezone.c -> build/lib.macosx-15.0-arm64-cpython-313/backports/datetime_fromisoformat
      copying backports/datetime_fromisoformat/timezone.h -> build/lib.macosx-15.0-arm64-cpython-313/backports/datetime_fromisoformat
      running build_ext
      building 'backports._datetime_fromisoformat' extension
      creating build/temp.macosx-15.0-arm64-cpython-313/backports/datetime_fromisoformat
      clang -fno-strict-overflow -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -O3 -Wall -I/Users/marcoz/prj/rp2/.venv/include -I/opt/homebrew/opt/python@3.13/Frameworks/Python.framework/Versions/3.13/include/python3.13 -c backports/datetime_fromisoformat/_datetimemodule.c -o build/temp.macosx-15.0-arm64-cpython-313/backports/datetime_fromisoformat/_datetimemodule.o
      backports/datetime_fromisoformat/_datetimemodule.c:795:25: error: call to undeclared function '_PyUnicode_Copy'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
          PyObject *str_out = _PyUnicode_Copy(dtstr);
                              ^
      backports/datetime_fromisoformat/_datetimemodule.c:795:15: error: incompatible integer to pointer conversion initializing 'PyObject *' (aka 'struct _object *') with an expression of type 'int' [-Wint-conversion]
          PyObject *str_out = _PyUnicode_Copy(dtstr);
                    ^         ~~~~~~~~~~~~~~~~~~~~~~
      2 errors generated.
      error: command '/usr/bin/clang' failed with exit code 1
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for backports-datetime-fromisoformat
Failed to build backports-datetime-fromisoformat
ERROR: ERROR: Failed to build installable wheels for some pyproject.toml based projects (backports-datetime-fromisoformat)
```

Any help would be appreciated, thanks.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -24,7 +24,8 @@ However, starting in version 2, `backports.datetime_fromisoformat` will apply it
 
 ## Unreleased
 
-* Nil.
+* Silenced errors when compiling against Python 3.13
+  * Note: `backports.datetime_fromisoformat` does nothing when used against Python 3.11+. Use `backports-datetime-fromisoformat; python_version < '3.11'` in your dependency list to avoid even downloading it in modern Pythons
 
 ## Version 2.0.2
 

--- a/README.rst
+++ b/README.rst
@@ -33,7 +33,7 @@ Quick Start
 
 .. code:: bash
 
-  pip install backports-datetime-fromisoformat
+  pip install "backports-datetime-fromisoformat; python_version < '3.11'"
 
 **Usage:**
 

--- a/backports/datetime_fromisoformat/_datetimemodule.c
+++ b/backports/datetime_fromisoformat/_datetimemodule.c
@@ -3,6 +3,8 @@
  * 037e9125527d4a55af566f161c96a61b3c3fd998)
  * It was then refreshed using the Python 3.11 contents present at
  * 27d8dc2c9d3de886a884f79f0621d4586c0e0f7a
+ * Finally, the `_PyUnicode_Copy` implementation was copied from
+ * c8749b578324ad4089c8d014d9136bc42b065343
  *
  * Since then, I have:
  *   - torn out all the functionality that doesn't matter to
@@ -10,6 +12,7 @@
  *   - switched calls to datetime creation to use the versions found in
  *     `PyDateTimeAPI`
  *   - made minor changes to make it compilable for older versions of Python.
+ *     - Including in-lining a copy of _PyUnicode_Copy
  *
  * Below is a copy of the Python 3.11 code license
  * (from https://docs.python.org/3/license.html):
@@ -747,6 +750,29 @@ time_fromisoformat(PyObject *tstr)
 invalid_string_error:
     PyErr_Format(PyExc_ValueError, "Invalid isoformat string: %R", tstr);
     return NULL;
+}
+
+PyObject *
+_PyUnicode_Copy(PyObject *unicode)
+{
+    Py_ssize_t length;
+    PyObject *copy;
+
+    if (!PyUnicode_Check(unicode)) {
+        PyErr_BadInternalCall();
+        return NULL;
+    }
+
+    length = PyUnicode_GET_LENGTH(unicode);
+    copy = PyUnicode_New(length, PyUnicode_MAX_CHAR_VALUE(unicode));
+    if (!copy)
+        return NULL;
+    assert(PyUnicode_KIND(copy) == PyUnicode_KIND(unicode));
+
+    memcpy(PyUnicode_DATA(copy), PyUnicode_DATA(unicode),
+           length * PyUnicode_KIND(unicode));
+    assert(_PyUnicode_CheckConsistency(copy, 1));
+    return copy;
 }
 
 static PyObject *

